modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/display/context/PortletConfigurationPermissionsDisplayContext.java

```diff
@@ -315,8 +328,14 @@ public String getResourcePrimKey() throws ResourcePrimKeyException {
 			return _resourcePrimKey;
 		}
 
-		_resourcePrimKey = ParamUtil.getString(
-			_httpServletRequest, "resourcePrimKey");
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-87806"))) {
+			getResourcePrimKeys();
+			_resourcePrimKey = _resourcePrimKeys[0];
```

Comment:
Can you use a local var to save the result of `getResourcePrimKeys()`? It is ugly to depend on the side effects of a getter.
```
long[] resourcePrimKeys = getResourcePrimKeys();

_resourcePrimKey = resourcePrimKeys[0];
```